### PR TITLE
Allow setting flutterSdkPath and dartSdkPath for the flutter and dart commands

### PR DESCRIPTION
**Pull request by passsy**
This change allows pinning the Flutter/Dart SDK using `flutterw`, `fvm` or other mechanisms. 
All that has to change is setting the path in `initializeSidekick()`

```dart
final runner = initializeSidekick(
  name: 'dash',
  // New
  dartSdkPath: 'path/to/dart/sdk',
  flutterSdkPath: 'path/to/flutter/sdk',
);
```

Both, `<cli> dart` and `<cli> flutter` now check for those paths and crash otherwise.

API changes:
- Added `dartSdkPath` and `flutterSdkPath` to `initializeSidekick` 
- The `flutterw` is now deprecated in favor of the new `flutter` function
- ~**Breaking:** `flutter()` and `dart()` now don't check for the `flutter_wrapper` `.flutter` dir~
- `flutterw()` and `dart()` now print a warning when using the deprecated `flutter_wrapper` is used implicitly  
- The dart and flutter commands now don't call `exit` directly, instead only set the exit code

## TODO

- [x] make `flutterw` change non breaking
- [x] make tests without network access

### CODE REVIEW

File: sidekick_core/lib/sidekick_core.dart [hunk 1]
```diff
@@ -34,13 +35,23 @@ export 'package:sidekick_core/src/sidekick_package.dart';
...
+/// [mainProjectPath] should be set when you have a package that you
+/// consider the main package of the whole repository.
+/// When your repository contains only one Flutter package in root set
+/// `mainProjectPath = '.'`.
+/// In a multi package repository you might use the same when the main package
+/// is in root, or `mainProjectPath = 'packages/my_app'` when it is in a subfolder.
+///
+/// Set [flutterSdkPath] when you bind a flutter sdk to this project. This SDK
+/// enables the [flutter] and [dart] commands.
+/// [dartSdkPath] is inherited from [flutterSdkPath]. Set it only for pure dart
```
Comment (Giuspepe) [inline sidekick_core/lib/sidekick_core.dart:47]: We could add a check that the user doesn't set both `dartSdkPath` and `flutterSdkPath` (`assert(dartSdkPath == null || flutterSdkPath == null)`

File: sidekick_core/test/dart_command_test.dart [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        // ignore: avoid_redundant_argument_values
+        dartSdkPath: null,
+      );
+      runner.addCommand(DartCommand());
+      try {
+        await runner.run(['dart']);
+        fail('did not throw');
+      } catch (e) {
+        expect(e, isA<DartSdkNotSetException>());
+      }
```
Comment (Giuspepe) [inline sidekick_core/test/dart_command_test.dart:32]: You can do this in just a single line
```suggestion
      expect(() => runner.run(['dart']), throwsA(isA<DartSdkNotSetException>()));
```

File: sidekick_core/test/fake_sdk.dart [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import 'package:dcli/dcli.dart' as dcli;
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:test/test.dart';
+
+/// Creates a fake Flutter SDK in temp with a `flutter` executable that does
+/// nothing besides "downloading" a fake Dart executable that does also nothing
+Directory fakeFlutterSdk() {
```
Comment (Giuspepe) [inline sidekick_core/test/fake_sdk.dart:7]: Nice!

File: sidekick_core/test/flutter_command_test.dart [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        // ignore: avoid_redundant_argument_values
+        flutterSdkPath: null,
+      );
+      runner.addCommand(FlutterCommand());
+      try {
+        await runner.run(['flutter']);
+        fail('did not throw');
+      } catch (e) {
+        expect(e, isA<FlutterSdkNotSetException>());
+      }
```
Comment (Giuspepe) [inline sidekick_core/test/flutter_command_test.dart:35]:
```suggestion
      expect(() => runner.run(['dart']), throwsA(isA<FlutterSdkNotSetException>()));
```

File: sidekick_core/test/flutter_command_test.dart [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          name: 'dash',
+          flutterSdkPath: fakeFlutterSdk().path,
+        );
+        runner.addCommand(FlutterCommand());
+        await runner.run(['flutter']);
+      });
+    },
+  );
+
+  test('flutter command fails when dartSdkPath is not set', () async {
```
Comment (Giuspepe) [inline sidekick_core/test/flutter_command_test.dart:22]:
```suggestion
  test('flutter command fails when flutterSdkPath is not set', () async {
```

File: sidekick_core/test/flutter_command_test.dart [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import 'package:sidekick_core/sidekick_core.dart';
+import 'package:test/test.dart';
+
+import 'fake_sdk.dart';
+import 'init_test.dart';
+
+void main() {
+  test(
+    'flutter command works when dartSdkPath is set',
```
Comment (Giuspepe) [inline sidekick_core/test/flutter_command_test.dart:9]:
```suggestion
    'flutter command works when flutterSdkPath is set',
```

**Review by Giuspepe — APPROVED** [review]
Works great :)
Maybe we should ask the user during `sidekick init` for a `dartSdkPath`/`flutterSdkPath` so they can directly use the `<cli> dart` and `<cli> flutter` commands without having to edit the `initializeSidekick` manually. I would do that in another PR though.

File: sidekick_core/lib/sidekick_core.dart (hunk 1 above)
Comment (passsy) [reply sidekick_core/lib/sidekick_core.dart:47]: Technically that shouldn't be a problem

File: sidekick_core/test/fake_sdk.dart (hunk 3 above)
Comment (passsy) [reply sidekick_core/test/fake_sdk.dart:7]: Old habits :)

File: sidekick_core/test/flutter_command_test.dart (hunk 4 above)
Comment (passsy) [reply sidekick_core/test/flutter_command_test.dart:35]: Old habits

**Pull request merged**